src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+			try {
+				msg = historyRef.getHttpMessage().cloneRequest();
+			} catch (HttpMalformedHeaderException | DatabaseException mhe){
+            			throw new IllegalArgumentException("Malformed header error.", mhe);
+			}
+			NameValuePair Original = null;
+			for (org.zaproxy.zap.model.NameValuePair parameter : listParam) {
+				if (field == parameter.getName()){
+					Original = parameter;
+					break;
+				}
+			}
+
+			//String[] list = {"101", "aghjjkkdn", "Finland", "???", "1", "das.anamika339@gmail.com", "377sjndbjkk@@@#"};
+			//Range for each characters set
+			int start=0;
+			int end=0;
+			
+			if ( !lowerR.getText().isEmpty() && !highR.getText().isEmpty() ) {
+				try {
+					start = Integer.parseInt(lowerR.getText());
+				} catch (NumberFormatException e) {
+					LOGGER.error(e.getMessage(), e);
+				}
+				try {
+					end = Integer.parseInt(highR.getText());
+				} catch (NumberFormatException e) {
+					LOGGER.error(e.getMessage(), e);
+				}	
+			}
+			else {
+				if (selected_chars == "US-ASCII") {
+					start = 0;
+					end = 128;
+				}
+				else if (selected_chars == "UTF-8") {
+					start = 0;
+					end = 513;
+				}	
+				else {
+					start =0;
+					end = 226;
+				}
+			}
+
+			for (int ch = start; ch < end; ch++) {
+    			  String chars =  Character.toString((char) ch);
+
+			  String letter;
+
+			if(pre!=null && post!=null){
+				letter = pre.concat(chars);
+				letter = letter.concat(post);
+			}
+			else if (pre!=null && post==null){
+				letter = pre.concat(chars);
+			}
+			else if (pre==null && post!=null){
+                                letter = chars.concat(post);
+                        }
+			else {
+				letter = chars;
+			}
+
+			
+			setParameter(msg, Original, field, letter);
+			msg.getRequestHeader().setContentLength(msg.getRequestBody().length());
+			try {
+                                gethttpSender().sendAndReceive(msg, false);
+                        }catch (IOException ioe) {
+                                throw new  IllegalArgumentException("IO error in sending request: " + ioe.getClass() + ": " + ioe.getMessage(), ioe);
+                        }
+			String response = msg.getResponseBody().toString();
+			System.out.println("Letter : " + letter + "Response : " + response);
+			Pattern pattern = Pattern.compile(regex.getText());
+			Matcher matcher = pattern.matcher(response);
+			if (matcher.find()) { 
+				model.addRow(new Object[]{letter, "FAILED"});
+				System.out.println("Match Found"); 
+				i_chars.append(letter).append(" \n");
+			}
+			else {
+				model.addRow(new Object[]{letter, "SUCCESS"});
+			}
+//msg.getResponseHeader().getStatusCode() + " " + msg.getResponseHeader().getReasonPhrase()); 
+			persistAndShowMessage(msg);
+			}
+			}
+
+		});
+		blacklist.addActionListener(new ActionListener () {
+		@Override
+			public void actionPerformed(ActionEvent event) {
+				JOptionPane.showMessageDialog(FieldEnumeration.this,"Illegal Characters: \n" + i_chars.toString());	
```

Comment:
i18n?
`whatever.key=Illegal Characters: \n {0}`
then use it:
`Constant.messages.getString("whatever.key",i_chars.toString());`

(Obviously whatever.key is a placeholder but you get the idea)